src/pages/iou/request/step/IOURequestStepCategory.tsx

```diff
@@ -0,0 +1,171 @@
+import type {StackScreenProps} from '@react-navigation/stack';
+import lodashIsEmpty from 'lodash/isEmpty';
+import React from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
+import {withOnyx} from 'react-native-onyx';
+import CategoryPicker from '@components/CategoryPicker';
+import type {ListItem} from '@components/SelectionList/types';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+import type {MoneyRequestNavigatorParamList} from '@libs/Navigation/types';
+import * as OptionsListUtils from '@libs/OptionsListUtils';
+import * as ReportUtils from '@libs/ReportUtils';
+import * as TransactionUtils from '@libs/TransactionUtils';
+import * as IOU from '@userActions/IOU';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import type {Policy, PolicyCategories, PolicyTagList, ReportActions, Session, Transaction} from '@src/types/onyx';
+import StepScreenWrapper from './StepScreenWrapper';
+import withFullTransactionOrNotFound from './withFullTransactionOrNotFound';
+import type {WithWritableReportOrNotFoundProps} from './withWritableReportOrNotFound';
+import withWritableReportOrNotFound from './withWritableReportOrNotFound';
+
+type IOURequestStepCategoryStackProps = {stackProps: StackScreenProps<MoneyRequestNavigatorParamList, typeof SCREENS.MONEY_REQUEST.STEP_CATEGORY>};
+
+type IOURequestStepCategoryOnyxProps = {
+    /** The draft transaction that holds data to be persisted on the current transaction */
+    splitDraftTransaction: OnyxEntry<Transaction>;
+    /** The policy of the report */
+    policy: OnyxEntry<Policy>;
+    /** Collection of categories attached to a policy */
+    policyCategories: OnyxEntry<PolicyCategories>;
+    /** Collection of tags attached to a policy */
+    policyTags: OnyxEntry<PolicyTagList>;
+    /** The actions from the parent report */
+    reportActions: OnyxEntry<ReportActions>;
+    /** Session info for the currently logged in user. */
+    session: OnyxEntry<Session>;
+};
+
+type IOURequestStepCategoryProps = IOURequestStepCategoryStackProps &
+    IOURequestStepCategoryOnyxProps &
+    WithWritableReportOrNotFoundProps & {
+        /** Holds data related to Money Request view state, rather than the underlying Money Request data. */
+        transaction: OnyxEntry<Transaction>;
+    };
+
+function IOURequestStepCategory({
+    report,
+    stackProps: {
```

Comment:
We don't use stackProps anywhere in the project, let's use `route` as before

Reply:
Also you might face the problem described [here](https://swmansion.slack.com/archives/C05P72RDZD2/p1711453240230669), please have a look